src/main/java/com/aws/greengrass/provisioning/ProvisioningHelper.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.provisioning;
+
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.config.UpdateBehaviorTree;
+import com.aws.greengrass.deployment.DeviceConfiguration;
+import com.aws.greengrass.lifecyclemanager.Kernel;
+import com.aws.greengrass.provisioning.ProvisionConfiguration.NucleusConfiguration;
+import com.aws.greengrass.provisioning.ProvisionConfiguration.SystemConfiguration;
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.aws.greengrass.componentmanager.KernelConfigResolver.CONFIGURATION_CONFIG_KEY;
+import static com.aws.greengrass.deployment.DeviceConfiguration.SYSTEM_NAMESPACE_KEY;
+import static com.aws.greengrass.lifecyclemanager.GreengrassService.SERVICES_NAMESPACE_TOPIC;
+
+@AllArgsConstructor
+@NoArgsConstructor
+public class ProvisioningHelper {
```

Comment:
We will now have ProvisioningHelper and an existing DeviceProvisioningHelper, should we be renaming one of them to avoid ambiguity? 
Alternately we could avoid adding this class at all if `SystemConfiguration` and `NucleusConfiguration` were to have a `toMap` method which would construct a config map and `KernelLifecycle` could directly invoke that `updateFromMap` with that

Reply:
Renamed to ProvisioningConfigUpdateHelper